### PR TITLE
fix: DVCPopulatedUser user_id and isAnonymous handling to be consistent with other SDKs

### PR DESCRIPTION
**Pull request by jonathannorris**
## Update DVCPopulatedUser constructor logic for anonymous user handling, and OpenFeature Web Provider anon user Contexts

This pull request introduces significant updates to the handling of anonymous users, alongside improvements to test coverage and error messaging. The most important changes include refining the logic for anonymous and non-anonymous user creation, updating test cases to reflect the new behavior, updating the OpenFeature Web Provider to handle anon Contexts, and enhancing the flexibility of user ID resolution in various contexts.

### User Creation Logic Updates:
* [`sdk/js/src/User.ts`](diffhunk://#diff-c043fbefa1714bfa4e7dc2c288591e7c668c1db171aa82a454c2b2eef00aaa97R39-R87): Enhanced the `DVCPopulatedUser` class to handle cases where `user_id` is an empty string or missing. Introduced a `generateAndSaveAnonUserId` method to create and optionally store anonymous user IDs. Improved validation to ensure `isAnonymous: false` requires a valid `user_id`.

* [`sdk/js/src/Client.ts`](diffhunk://#diff-7e88ae7eb811449dfe4a792952dab7d8accab565833b6a8a5764488a232d4325R210-R211): Updated the `DevCycleClient` to remove redundant logic for saving anonymous user IDs and ensure proper handling of anonymous user creation during initialization and user updates. 

### Test Coverage Enhancements:
* [`sdk/js/__tests__/User.spec.js`](diffhunk://#diff-14fe0afb5745007258545fb9cc3d78caece8562444a2be2c765b0ef63b5052beL61-R109): Overhauled test cases for `DVCPopulatedUser` to verify the new behaviour for anonymous and non-anonymous users, including edge cases like empty strings and whitespace-only `user_id` values.

* [`sdk/openfeature-web-provider/__tests__/DevCycleProvider.test.ts`](diffhunk://#diff-e58d167eddde158b90ed1d76bf8c728bb1d1505c7b10a943047e1dc1be2fdf04R142-R232): Added tests for fallback logic when resolving `user_id` from `targetingKey`, `userId`, or other sources. Verified behaviour when no valid user ID is provided and when `isAnonymous` is explicitly set. [[1]](diffhunk://#diff-e58d167eddde158b90ed1d76bf8c728bb1d1505c7b10a943047e1dc1be2fdf04R142-R232) [[2]](diffhunk://#diff-e58d167eddde158b90ed1d76bf8c728bb1d1505c7b10a943047e1dc1be2fdf04L172-R261)

### Error Messaging Improvements:
* [`sdk/js/__tests__/Client.spec.js`](diffhunk://#diff-e690cf47bd486dffaa20b9313392a6cb432489fc90c59293e8e9deda7e58f2d1L697-R701): Updated error messages in `DevCycleClient` tests to reflect the new validation rules for `isAnonymous` and `user_id`.

* [`sdk/openfeature-web-provider/__tests__/DevCycleProvider.test.ts`](diffhunk://#diff-e58d167eddde158b90ed1d76bf8c728bb1d1505c7b10a943047e1dc1be2fdf04L204-R292): Improved warnings for invalid `EvaluationContext` properties, including mismatched types for `isAnonymous` and `customData`.

### CODE REVIEW

File: sdk/js/__tests__/User.spec.js [hunk 1]
```diff
@@ -58,29 +58,47 @@ describe('DVCPopulatedUser tests', () => {
...
         const newUser = new DVCPopulatedUser({ isAnonymous: true })
         expect(uuidValidate(newUser.user_id)).toBe(true)
+        expect(newUser.isAnonymous).toBe(true)
     })
 
-    it('should throw an error if the user id is an empty string', () => {
+    it('should use provided user_id when isAnonymous is true', () => {
+        const user = new DVCPopulatedUser({ user_id: 'abc123', isAnonymous: true })
+        expect(user.user_id).toBe('abc123')
+        expect(user.isAnonymous).toBe(true)
```
Comment (JamieSinn) [inline sdk/js/__tests__/User.spec.js:76]: how does calling resetuser change this?

Comment (jonathannorris) [reply sdk/js/__tests__/User.spec.js]: what do you mean? This is all kinda outside of `resetUser`: 
image

Comment (JamieSinn) [reply sdk/js/__tests__/User.spec.js]: More the concept that if you have an anon user and you're setting a value for the user id - do we retain that or wipe it?

Comment (JamieSinn) [reply sdk/js/__tests__/User.spec.js]: more a curiosity thing than anything

Comment (jonathannorris) [reply sdk/js/__tests__/User.spec.js]: good point I think that would be an issue right now with how its setup: 
image

**Review by JamieSinn — APPROVED** [review]
